src/types/UiaTextRangeBase.cpp

```diff
@@ -1522,6 +1522,12 @@ void UiaTextRangeBase::_moveEndpointByUnitWord(_In_ const int moveCount,
             {
                 success = false;
             }
+            else if (const auto wordStart{ buffer.GetWordStart(nextPos, _wordDelimiters, true, documentEnd) }; allowBottomExclusive && nextPos != wordStart)
+            {
+                // Degenerate ranges first move to the beginning of the word
+                resultPos = wordStart;
+                (*pAmountMoved)--;
+            }

```

Comment:
Ok, this is really annoying, but I can't think of an elegant way to fix this. Right now, we're _always_ getting `wordStart`, but then if `allowBottomExclusive` is `false`, we just throw it away and never use it.

Is there a trick we can use to still short-circuit this?

I basically want something like this:
```cpp
else if (allowBottomExclusive &&
          /* NOW calculate wordStart because we're actually gonna use it */
          (const auto wordStart{ buffer.GetWordStart(nextPos, _wordDelimiters, true, documentEnd) };
           nextPos != wordStart))
{
    resultPos = wordStart;
    (*pAmountMoved)--;
}
```

Reply:
A function?

```
else if (allowBottomExclusive && !_isNextPosWordStart(buffer, nextPos, _wordDelimiters, true, documentEnd, &resultPos))
```